home-depot-vanilla/website.js

```diff
@@ -0,0 +1,125 @@
+var allDepItems = document.querySelector(".all-departments-item")
+var allDepSubDrop = document.querySelector(".all-departments-subdropdown")
+//All Departments subdropdown logic
+    allDepItems.addEventListener('mouseenter', () => {
+        allDepSubDrop.style.display = 'block'
+    })
+    allDepItems.addEventListener('mouseleave', () => {
+        allDepSubDrop.style.display = 'none'
+    })
+    allDepSubDrop.addEventListener('mouseenter', () => {
+        allDepSubDrop.style.display = 'block'
+    })
+    allDepSubDrop.addEventListener('mouseleave', () => {
+        allDepSubDrop.style.display = 'none'
+    })
+var HDFK = document.querySelector(".HD-FK")
+var HDFKdropdown = document.querySelector(".HD-FK-dropdown")
+//HDFK dropdown logic
+    HDFK.addEventListener('mouseenter', () => {
+        HDFKdropdown.style.display = 'flex'
+    })
+    HDFK.addEventListener('mouseleave', () => {
+        HDFKdropdown.style.display = 'none'
+    })
+    HDFKdropdown.addEventListener('mouseenter', () => {
+        HDFKdropdown.style.display = 'flex'
+        HDFK.style.fontWeight = 'bold';
+    })
+    HDFKdropdown.addEventListener('mouseleave', () => {
+        HDFKdropdown.style.display = 'none'
+        HDFKbolean = false;
+        HDFK.style.fontWeight = 'normal';
+    })
+var HDFK = document.querySelector(".HD-FK")
+var HDFKdropdown = document.querySelector(".HD-FK-dropdown")
+//HDFK dropdown logic
+    HDFK.addEventListener('mouseenter', () => {
+        HDFKdropdown.style.display = 'flex'
+    })
+    HDFK.addEventListener('mouseleave', () => {
+        HDFKdropdown.style.display = 'none'
+    })
+    HDFKdropdown.addEventListener('mouseenter', () => {
+        HDFKdropdown.style.display = 'flex'
+        HDFK.style.fontWeight = 'bold';
+    })
+    HDFKdropdown.addEventListener('mouseleave', () => {
+        HDFKdropdown.style.display = 'none'
+        HDFK.style.fontWeight = 'normal';
+    })
+var homeDecor = document.querySelector(".home-decor")
+var HDFKsubdropdown = document.querySelector(".HD-FK-subdropdown")
+    //HDFK subdropdown logic
+    homeDecor.addEventListener('mouseenter', () => {
+        HDFKsubdropdown.style.display = 'flex'
+    })
+    homeDecor.addEventListener('mouseleave', () => {
+        HDFKsubdropdown.style.display = 'none'
+    })
+    HDFKsubdropdown.addEventListener('mouseenter', () => {
+        HDFKsubdropdown.style.display = 'flex'
+        HDFKdropdown.style.display = 'flex'
+    })
+    HDFKsubdropdown.addEventListener('mouseleave', () => {
+        HDFKsubdropdown.style.display = 'none'
+        HDFKdropdown.style.display = 'none'
+
+    })
+var DIYprojects = document.querySelector(".DIY-projects")
+var DIYdrop = document.querySelector(".DIY-dropdown")
+    //DIY projects dropdown logic
+    DIYprojects.addEventListener('mouseenter', () => {
```

Comment:
Can you think of ways to refactor these event listeners to reuse code? Some of these functions repeat.